packages/block-library/src/heading/edit.js

```diff
@@ -37,6 +41,33 @@ function HeadingEdit( {
 		style,
 	} );
 
+	// Initially set anchor for headings that have content but no anchor set.
+	// This is used when transforming a block to heading, or for legacy anchors.
+	useEffect( () => {
+		if ( ! anchor && content ) {
+			setAttributes( {
+				anchor: generateAnchor( clientId, content, allHeadingAnchors ),
+			} );
+		}
+		allHeadingAnchors[ clientId ] = anchor;
```

Comment:
If you need to set a key&value pair, the we should use `Map` here, see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map.

Reply:
Out of curiosity: what's wrong with using an object here?

Reply:
I guess it's fine to use a plain object, too. It all depends on how complex operations need to be performed here.

Reply:
Since we're just doing pretty basic operations and simply adding/deleting key/value pairs, an object may make more sense... Changed it to use an object instead of a `Set` :+1: